### DISCUSSION TITLE
A way to dig a polygon on the imageLayer except with TerrainConstraintLayer？

### DISCUSSION THREAD

**Discussion by luoxiaozhu** [pelicanmapping/osgearth#2542] [Support Q&A] [answered]
i added two xyzImageLayers on the earth,and i want to dig a polygon on the lower xyzImageLayer， i know TerrainConstraintLayer works，but i don‘t want to dig my ElevationLayer. So,is there a way to dig a polygon on the imageLayer except with TerrainConstraintLayer? i have try to make a custom shader,code like this:
```c++
const char* fragShader = R"(
        #version 330
        void make_it_red(inout vec4 color) {
			vec4 colorVec4 = color - vec4(1.0,1.0,1.0,1.0);
			if(length(colorVec4)<0.00001)
			{				
				color.a = 0.0;
            }
        }
    )";
VirtualProgram* vp = VirtualProgram::getOrCreate(mapNode->getOrCreateStateSet());
vp->setFunction("make_it_red", fragShader, VirtualProgram::LOCATION_FRAGMENT_LIGHTING);
mapNode->getOrCreateStateSet()->setAttributeAndModes(vp, osg::StateAttribute::ON);
```

but It only works on the upper ImageLayer

**Comment by gwaldron** [comment]
I do not understand what you want to see. Can you explain it a different way?

**Reply by luoxiaozhu** [reply]
Thanks for your response! I have a fully transparent image loaded with a custom ImageLayer and I already have an image layer loaded with xyzImageLayer on my earth and I want the xyzImageLayer overlaid part by the custom imageLayer to be fully transparent as well.
![image](https://github.com/gwaldron/osgearth/assets/5326405/d04cea49-8238-44c7-beb8-910b289ecc6c)
![image](https://github.com/gwaldron/osgearth/assets/5326405/f91e8e83-68a2-43a2-a2c7-9cc764bbc6d1)
the first picture show what i have seen so far,and the second picture show what i want  to see which i made by TerrainConstraintLayer,I don't want to use TerrainConstraintLayer because I might have to hide and show this area frequently,thank you again!

**Reply by gwaldron** [reply, marked as answer]
Take a look at the example `shared.earth` in the `tests` folder; this concept lets you access one layer's textures in another layer's shaders.

**Reply by luoxiaozhu** [reply]
Thank you gwaldron，that is what i want！
